plugin/kubernetes/external.go

```diff
@@ -75,16 +85,52 @@ func (k *Kubernetes) External(state request.Request) ([]msg.Service, int) {
 			continue
 		}
 
-		for _, ip := range svc.ExternalIPs {
-			for _, p := range svc.Ports {
-				if !(matchPortAndProtocol(port, p.Name, protocol, string(p.Protocol))) {
+		if svc.Headless() || endpoint != "" {
+			if endpointsList == nil {
+				endpointsList = endpointsListFunc()
```

Comment:
Is the function variable indirection necessary?  Can you just call `k.APIConn.EpIndex(idx)` directly?

Reply:
Yep, it's not cecessary, removed.